example-playbooks/async_extra/action_plugins/async_task.py

```diff
@@ -202,6 +273,56 @@ def run_async_task(self, vars):
         result = handler.run(task_vars=vars)
         return self.v2_on_result(Host(), task, result)
 
+    # derived from ansible.legacy.script action plugin.
+    # noinspection PyProtectedMember
+    def prepare_script(self):
+        # executable, creates, removes are all handled by the ansible.legacy.command module.
+        # become is handled by the self._execute_module() part of the "command" action plugin.
+        #
+        # Split out the script as the first item in command using
+        # shlex.split() in order to support paths and files with spaces in the name.
+        # Any arguments passed to the script will be added back later.
+        command = to_native(self._task.args.get('script', ''), errors='surrogate_or_strict')
+        parts = [to_text(s, errors='surrogate_or_strict') for s in shlex.split(command.strip())]
+        source = parts[0]
+
+        try:
+            source = self._loader.get_real_file(self._find_needle('files', source), decrypt=self._task.args.get('decrypt', True))
+        except AnsibleError as e:
+            raise AnsibleActionFail(to_native(e))
+
+        if self._play_context.check_mode:
+            raise _AnsibleActionDone()
+
+        # transfer the file to a remote tmp location
+        tmp_src = self._connection._shell.join_path(self._connection._shell.tmpdir,
+                                                    os.path.basename(source))
+
+        # Convert command to text for the purpose of replacing the script since
+        # parts and tmp_src are both unicode strings and command will be different
+        # depending on Python version.
+        #
+        # Once everything is encoded consistently, replace the script path on the remote
+        # system with the remainder of the command. This preserves quoting in parameters
+        # that would have been removed by shlex.split().
+        target_command = to_text(command).strip().replace(parts[0], tmp_src)
+
+        self._transfer_file(source, tmp_src)
+
+        # set file permissions, more permissive when the copy is done as a different user
+        self._fixup_perms2((self._connection._shell.tmpdir, tmp_src), execute=True)
```

Comment:
What if this command fails (e.g. you don't have enough permissions). What will cleanup the state (e.g. delete the file transferred in the like 310 above)?

Reply:
nothing. async jobs do not cleanup tmp folders.

Reply:
Wouldn't it cause problems if one wants to re-run the command?

Reply:
no